cvector.h

```diff
@@ -0,0 +1,45 @@
+typedef struct
+{
+    char *data;
```

Comment:
The data is stored as a char*. But when you get elements, it will return a void*. It's the user's job to keep track of the element type and cast it appropriately